src/proposals/20241114/SparkEthereum_20241114.t.sol

```diff
@@ -0,0 +1,549 @@
+
+import { IALMProxy }         from "spark-alm-controller/src/interfaces/IALMProxy.sol";
+import { IRateLimits }       from "spark-alm-controller/src/interfaces/IRateLimits.sol";
+import { MainnetController } from "spark-alm-controller/src/MainnetController.sol";
+import { ForeignController } from "spark-alm-controller/src/ForeignController.sol";
+import { RateLimitHelpers }  from "spark-alm-controller/src/RateLimitHelpers.sol";
+
+interface IVatLike {
+    function ilks(bytes32) external view returns (uint256, uint256, uint256, uint256, uint256);
+}
+
+interface IPSMLike {
+    function convertToShares(address, uint256) external view returns (uint256);
+    function convertToAssetValue(uint256) external view returns (uint256);
+    function shares(address) external view returns (uint256);
+    function swapExactIn(
+        address assetIn,
+        address assetOut,
+        uint256 amountIn,
+        uint256 minAmountOut,
+        address receiver,
+        uint256 referralCode
+    ) external returns (uint256 amountOut);
+}
+
+interface IVariableKinkIRM {
+    function getVariableRateSlope1Spread() external view returns (int256);
+}
+
+contract SparkEthereum_20241114Test is SparkEthereumTestBase {
+
+    using DomainHelpers         for *;
+    using OptimismBridgeTesting for *;
+
+    address internal constant PT_26DEC2024_PRICE_FEED  = 0x81E5E28F33D314e9211885d6f0F4080E755e4595;
+    address internal constant PT_SUSDE_26DEC2024       = 0xEe9085fC268F6727d5D4293dBABccF901ffDCC29;
+
+    address internal constant PT_27MAR2025_PRICE_FEED  = 0x38d130cEe60CDa080A3b3aC94C79c34B6Fc919A7;
+    address internal constant PT_SUSDE_27MAR2025       = 0xE00bd3Df25fb187d6ABBB620b3dfd19839947b81;
+
+    address internal constant OLD_WETH_INTEREST_RATE_STRATEGY = 0x6fd32465a23aa0DBaE0D813B7157D8CB2b08Dae4;
+    address internal constant NEW_WETH_INTEREST_RATE_STRATEGY = 0xf4268AeC16d13446381F8a2c9bB05239323756ca;
+
+    uint256 internal constant SUSDS_DEPOSIT_AMOUNT = 8_000_000e18;
+    uint256 internal constant USDS_BRIDGE_AMOUNT   = 1_000_000e18;
+
+    address internal constant FREEZER = 0x90D8c80C028B4C09C0d8dcAab9bbB057F0513431;  // Gov. facilitator multisig
+    address internal constant RELAYER = 0x8a25A24EDE9482C4Fc0738F99611BE58F1c839AB;  // Same address on all chains
+
+    address internal constant DEPLOYER = 0x6F3066538A648b9CFad0679DF0a7e40882A23AA4;
+
+    Domain mainnet;
+    Domain base;
+
+    Bridge nativeBridge;
+    Bridge cctpBridge;
+
+    address basePayload;
+
+    constructor() {
+        id = '20241114';
+    }
+
+    function setUp() public {
+        mainnet = getChain('mainnet').createFork(21122303);  // Nov 5, 2024
+        base    = getChain('base').createFork(22015320);     // Nov 5, 2024
+
+        mainnet.selectFork();
+        payload = deployPayload();
+
+        // TODO replace this with the actual address when deployed
+        base.selectFork();
+        basePayload = deployPayloadBase();
```

Comment:
[In the last spell with two chains](https://github.com/marsfoundation/spark-spells/blob/33a274f5e0c268c4c812d58b82d5c715db9e794a/src/proposals/20240627/SparkEthereum_20240627.t.sol) (Ethereum and Gnosis) this part was a bit different 
```
        mainnet = new Domain(getChain('mainnet'));
        gnosis  = new GnosisDomain(getChain('gnosis_chain'), mainnet);

        mainnet.rollFork(20118549);  // June 18, 2024
        gnosis.rollFork(34526700);   // June 18, 2024

        gnosis.selectFork();
        new SparkGnosis_20240627();

        mainnet.selectFork();
        payload = deployPayload();
``` 
Why is that?

Reply:
Not sure why that was done - maybe a mistake? It doesn't make sense to me. The end result is something different anyways if you check the final version in the archive.

Reply:
`new Domain()` and `rollFork()` are still there. In general, it's ok, just was curious why it's different